pkg/validator/validator.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validator
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/okteto/okteto/pkg/env"
+)
+
+// ErrReservedVariableName is raised when a variable from cmd option has invalid name
+var ErrReservedVariableName = errors.New("reserved variable name. See documentation for more info: https://www.okteto.com/docs/core/credentials/environment-variables/")
+
+// CheckReservedVariablesNameOption returns an error when any of the variable names from command flags are not allowed as input
+func CheckReservedVariablesNameOption(variables []string) error {
+	for _, v := range variables {
+		name, _, ok := strings.Cut(v, "=")
+		if ok && isReservedVariableName(name) {
+			return fmt.Errorf("%s is %w", name, ErrReservedVariableName)
```

Comment:
Could we use a `UserError` here? I think it's more readable if we had the `See documentation for more info: https://www.okteto.com/docs/core/credentials/environment-variables/` message in a new line

Reply:
I've updated this, it shows like this 

![Screenshot 2024-06-12 at 12 15 21](https://github.com/okteto/okteto/assets/40767058/7b01203b-c4e3-4f7f-99a5-eb4e3f8bea3d)